### PR TITLE
fix free on not initialised PubNub

### PR DESCRIPTION
**Pull request by Xavrax**
fix: fix `pubnub_free()` function on not initialised PubNub
Fix `pubnub_free()` function on not initialised PubNub that can cause exceptions/undefined behaviours

### CODE REVIEW

File: core/pubnub_alloc_std.c [hunk 1]
```diff
@@ -135,12 +135,18 @@ void pballoc_free_at_last(pubnub_t* pb)
...
-    int result = -1;
-
     PUBNUB_ASSERT(pb_valid_ctx_ptr(pb));
 
     PUBNUB_LOG_TRACE("pubnub_free(%p)\n", pb);
 
+    if (pb->state == PBS_NULL) {
+        PUBNUB_LOG_TRACE("pubnub_free(%p) not initialized - freeing...\n", pb);
+        remove_allocated(pb);
+        free(pb);
```
Comment (parfeon) [inline core/pubnub_alloc_std.c:145]: Wouldn't there be over release with `remove_allocated`?

Comment (Xavrax) [reply core/pubnub_alloc_std.c:145]: As i understood -`remove_allocated` only removes concrete instance from the memory. Nothing more.

Comment (Xavrax) [reply core/pubnub_alloc_std.c:145]: Also it frees only the whole list of contextes if there is nothing left.

**Review by parfeon — APPROVED** [review]
LGTM!

**Pull request merged**